# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Naeem34245/FlyRank-ML-INTERN/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Refresh / Content Opportunity Scoring (Lane 2)**

I'm choosing this lane because I already ran the starter pipeline (Notebook 01) and 
saw a random forest model reach Precision@50 = 0.740, clearly beating the baseline 
rule score of 0.240 — meaning a learned ranking finds nearly 3x more true opportunities 
in the same top-50 list. This lane also produces reason codes for every recommendation, 
which fits my interest in explainable, auditable model outputs rather than black-box scores.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:** Which content pages should FlyRank's review team look at first 
when they have limited time to act on content refresh?

**Unit of analysis:** one page (one content_id) at one point in time.

**Decision it improves:** which pages a content reviewer opens first, out of hundreds 
of candidates, given limited review capacity.

**Action someone takes:** a reviewer opens the top-ranked pages and decides to refresh, 
expand, protect, prune, or monitor each one, guided by the reason codes attached to it.

**Cost of a wrong recommendation:** 
- False positive (page flagged but not actually a problem): reviewer wastes limited time 
  on a page that didn't need attention, while a real problem page waits.
- False negative (a real declining/stale page not surfaced): the page keeps losing 
  visibility or traffic silently, and no one notices until it's much worse.

**Why data/ML helps here (not just intuition):** with tens of thousands of pages and only 
a handful of reviewer-hours per week, no human can manually check every page. A ranked, 
evidence-backed queue turns an impossible manual task into a short, prioritized list — 
and a learned model can combine many weak signals (position, CTR, freshness, engagement) 
in ways a single fixed rule can't.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv('../../data/raw/content_refresh_anonymized.csv')

total_pages = len(df)
declining_pages = (df['trend_direction'] == 'down').sum()
stale_visible = ((df['days_since_last_update'] >= 180) & (df['impressions_90d'] >= 500)).sum()

print(f"Total pages in starter dataset: {total_pages}")
print(f"Pages currently declining: {declining_pages} ({declining_pages/total_pages:.1%})")
print(f"Stale but still visible pages (>=180 days old, >=500 impressions): {stale_visible} ({stale_visible/total_pages:.1%})")

Total pages in starter dataset: 30000
Pages currently declining: 16262 (54.2%)
Stale but still visible pages (>=180 days old, >=500 impressions): 17 (0.1%)


Loading the starter dataset shows 30,000 total pages, of which 16,262 (54.2%) are 
currently declining — a large, real candidate pool for a review queue. The 
stale-but-visible count is small in this particular slice (17 pages, 0.1%, using a 
strict >=180-day / >=500-impression threshold), but the declining-pages number alone 
already shows this lane has plenty of signal worth ranking over the next 7 weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can claim:** this analysis can rank pages by evidence of decline, staleness, 
or opportunity, using observed signals — and can produce a ranked review queue that 
beats a simple fixed rule on standard ranking metrics (Precision@K).

**What I cannot claim:** 
- That refreshing a page WILL cause traffic to recover — that requires a causal 
  experiment, not observational data.
- That any single feature is "the reason" Google ranks a page a certain way — I have 
  no access to Google's algorithm.
- That the starter dataset result (Precision@50 = 0.740) will hold at full warehouse 
  scale — it was measured on a 30,000-row anonymized slice with client-holdout 
  validation, and needs to be re-earned on the larger release.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.